### PR TITLE
Journal voucher dev

### PR DESCRIPTION
**Pull request by prateek-giddh**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/accounting/voucher-grid/voucher-grid.component.html [hunk 1]
```diff
@@ -4,19 +4,12 @@
             <div class="clearfix pdT2">
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/accounting/voucher-grid/voucher-grid.component.html:4]: Deprecated file will be deleted once JV is complete (IGNORE)

File: apps/web-giddh/src/app/accounting/voucher-grid/voucher-grid.component.ts [hunk 2]
```diff
@@ -9,14 +9,13 @@ import { LedgerActions } from './../../actions/ledger/ledger.actions';
 import { IOption } from './../../theme/ng-select/option.interface';
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/accounting/voucher-grid/voucher-grid.component.ts:9]: Deprecated file will be deleted once JV is complete (IGNORE)

File: apps/web-giddh/src/app/accounting/journal-voucher/invoice/invoice.component.html [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+                    <label>Party A/c Name: <sup>*</sup></label>
+                    <!-- #creditor -->
+                    <input #partyAccNameInputField name="party_name" type="text" placeholder="Party account"
+                        class="form-control max-height28 appearance-none upper-fields"
+                        (input)="onPartyAccFocusInput($event);selectedField='account';focusedField = 'partyAcc';"
+                        (keyup)="searchAccount($event?.target?.value)" (keydown)="detectKey($event)"
+                        [(ngModel)]="creditorAcc.name"
+                        (focus)="onPartyAccFocus($event);selectedField='account';focusedField = 'partyAcc';"
+                        (blur)="onPartyAccBlur();selectedField=null;focusedField = null;filterByText = ''"
+                        style="max-width:200px;" />
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/accounting/journal-voucher/invoice/invoice.component.html:44]: remove inline css

File: apps/web-giddh/src/app/accounting/journal-voucher/invoice/invoice.component.html [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+            </div>
+            <!-- /party acc name -->
+
+            <!--  purchase as INVOICE -->
+            <div class="clearfix mrT1" [style.min-height.vh]="61">
+                <div class="table-responsive bdrT">
+                    <table class="table entry_table">
+                        <thead>
+                            <tr>
+                                <th style="width:70%;">Name of Item</th>
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/accounting/journal-voucher/invoice/invoice.component.html:66]: avoid inline css

File: apps/web-giddh/src/app/accounting/journal-voucher/invoice/invoice.component.html [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+                                    <input type="text" [value]="stockTotal" readonly disabled
+                                        class="form-control text-right" />
+                                    <div class="bdrB"></div>
+                                </td>
+                            </tr>
+                            <br />
+                            <!-- accounts entry -->
+                            <!--  -->
+                            <ng-container *ngFor="let transaction of accountsTransaction;let i=index;let l=last">
+                                <tr [ngClass]="{'activeRow': isSelectedRow && selectedAccIdx === i}">
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/accounting/journal-voucher/invoice/invoice.component.html:139]: rename index 'i'

File: apps/web-giddh/src/app/accounting/journal-voucher/journal-voucher.component.ts [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+    }
+
+    @HostListener('document:keydown', ['$event'])
+    public beforeunloadHandler(event: KeyboardEvent) {
+        return event.key !== FUNCTIONAL_KEYS.F5;
+    }
+
+    @HostListener('document:keyup', ['$event'])
+    public handleKeyboardEvent(event: KeyboardEvent) {
+        if (event.ctrlKey && event.key.toLowerCase() === 'a') { // Ctrl + A
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/accounting/journal-voucher/journal-voucher.component.ts:166]: nullity check for event

File: apps/web-giddh/src/app/accounting/journal-voucher/journal-voucher.component.ts [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+        let stateDetailsRequest = new StateDetailsRequest();
+        stateDetailsRequest.companyUniqueName = companyUniqueName;
+        stateDetailsRequest.lastState = 'journal-voucher';
+
+        this.store.dispatch(this.companyActions.SetStateDetails(stateDetailsRequest));
+
+        this.store.pipe(select(appState => appState.session.companyUniqueName), take(1)).subscribe(a => {
+            if (a && a !== '') {
+                this.accountService.getFlattenAccounts('', '', '').pipe(takeUntil(this.destroyed$)).subscribe(data => {
+                    if (data.status === 'success') {
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/accounting/journal-voucher/journal-voucher.component.ts:248]: nullity check for data

File: apps/web-giddh/src/app/accounting/journal-voucher/journal-voucher.component.ts (hunk 6 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/accounting/journal-voucher/journal-voucher.component.ts:166]: @rohit-walkover Not required as host listener is used and no custom calls are made

File: apps/web-giddh/src/app/accounting/journal-voucher/journal-voucher.component.ts (hunk 7 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/accounting/journal-voucher/journal-voucher.component.ts:248]: @rohit-walkover Done

File: apps/web-giddh/src/app/accounting/journal-voucher/invoice/invoice.component.html (hunk 5 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/accounting/journal-voucher/invoice/invoice.component.html:139]: @rohit-walkover This code is copy-paste and work will be done in near future on this

File: apps/web-giddh/src/app/accounting/journal-voucher/invoice/invoice.component.html (hunk 4 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/accounting/journal-voucher/invoice/invoice.component.html:66]: @rohit-walkover This code is copy-paste and work will be done in near future on this

File: apps/web-giddh/src/app/accounting/journal-voucher/invoice/invoice.component.html (hunk 3 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/accounting/journal-voucher/invoice/invoice.component.html:44]: @rohit-walkover This code is copy-paste and work will be done in near future on this

**Pull request merged**